PGIP/Server.hs

```diff
@@ -1806,27 +1822,224 @@ consNode :: LibEnv -> LibName -> DGraph -> (Int, DGNodeLab)
   -> G_sublogics -> Bool -> Maybe String -> Maybe String -> Maybe Int
   -> ResultT IO (LibEnv, [ProofResult])
 consNode le ln dg nl@(i, lb) subL useTh mp mt tl = do
-  consList <- lift $ getFilteredConsCheckers mt subL
+  (cc, c) <- liftIO $ findConsChecker mt subL mp
+  lift $ do
+    cstat <- consistencyCheck useTh cc c ln le dg nl $ fromMaybe 1 tl
+    -- Consistency Results are stored in LibEnv via DGChange object
+    let cSt = sType cstat
+        le'' = if cSt == CSUnchecked then le else
+               Map.insert ln (changeDGH dg $ SetNodeLab lb
+                 (i, case cSt of
+                       CSInconsistent -> markNodeInconsistent "" lb
+                       CSConsistent -> markNodeConsistent "" lb
+                       _ -> lb)) le
+    return (le'', [(" ", drop 2 $ show cSt, show cstat,
+                    AbsState.ConsChecker cc, c, Nothing, Just $ sMessage cstat)])
+
+findConsChecker :: Maybe String -> G_sublogics -> Maybe String
+                -> IO (G_cons_checker, AnyComorphism)
+findConsChecker translationM gSublogic consCheckerNameM = do
+  consList <- getFilteredConsCheckers translationM gSublogic
   let findCC x = filter ((== x ) . getCcName . fst) consList
-      mcc = maybe (findCC "darwin") findCC mp
-  case mcc of
+      consCheckersL = maybe (findCC "darwin") findCC consCheckerNameM
+  case consCheckersL of
         [] -> fail "no cons checker found"
-        ((cc, c) : _) -> lift $ do
-          cstat <- consistencyCheck useTh cc c ln le dg nl $ fromMaybe 1 tl
-          -- Consistency Results are stored in LibEnv via DGChange object
-          let cSt = sType cstat
-              le'' = if cSt == CSUnchecked then le else
-                     Map.insert ln (changeDGH dg $ SetNodeLab lb
-                       (i, case cSt of
-                             CSInconsistent -> markNodeInconsistent "" lb
-                             CSConsistent -> markNodeConsistent "" lb
-                             _ -> lb)) le
-          return (le'', [(" ", drop 2 $ show cSt, show cstat,
-                          AbsState.ConsChecker cc, c, Nothing)])
-
-proveNode :: LibEnv -> LibName -> DGraph -> (Int, DGNodeLab) -> G_theory
-  -> G_sublogics -> Bool -> Maybe String -> Maybe String -> Maybe Int
-  -> [String] -> [String] -> ResultT IO (LibEnv, [ProofResult])
+        (gConsChecker, comorphism) : _ -> return (gConsChecker, comorphism)
+
+reasonREST :: HetcatsOpts -> LibEnv -> LibName -> DGraph -> ProverMode
+           -> String -> ReasoningParameters
+           -> ResultT IO (LibEnv, [(String, [ProofResult])])
+reasonREST opts libEnv libName dGraph_ proverMode location reasoningParameters = do
+  reasoningCacheE <- liftIO $ buildReasoningCache
+  failOnLefts reasoningCacheE
+  let reasoningCache1 = map unRight $ filter (not . isLeft) reasoningCacheE
```

Comment:
shorter: `let reasoningCache1 = rights reasoningCacheE`
(you already know that there are only `Right` elements)